programs/drift/src/state/perp_market.rs

```diff
@@ -459,14 +461,43 @@ impl PerpMarket {
             MarginRequirementType::Maintenance => margin_ratio_maintenance,
         };
 
-        let size_adj_margin_ratio = calculate_size_premium_liability_weight(
-            size,
-            self.imf_factor,
-            default_margin_ratio,
-            MARGIN_PRECISION_U128,
-        )?;
-
-        let margin_ratio = default_margin_ratio.max(size_adj_margin_ratio);
+        let margin_ratio =
+            if is_high_leverage_user && margin_type != MarginRequirementType::Maintenance {
+                // use HLM maintenance margin but ordinary mode initial/fill margin for size adj calculation
+                let pre_size_adj_margin_ratio = match margin_type {
+                    MarginRequirementType::Initial => self.margin_ratio_initial,
+                    MarginRequirementType::Fill => {
+                        self.margin_ratio_initial
+                            .safe_add(self.margin_ratio_maintenance)?
+                            / 2
+                    }
+                    MarginRequirementType::Maintenance => margin_ratio_maintenance,
+                };
+
+                let size_adj_margin_ratio = calculate_size_premium_liability_weight(
+                    size,
+                    self.imf_factor,
+                    pre_size_adj_margin_ratio,
+                    MARGIN_PRECISION_U128,
+                    false,
+                )?;
+
+                calc_high_leverage_mode_initial_margin_ratio_from_size(
+                    pre_size_adj_margin_ratio,
+                    size_adj_margin_ratio,
+                    default_margin_ratio,
+                )?
+            } else {
+                let size_adj_margin_ratio = calculate_size_premium_liability_weight(
+                    size,
+                    self.imf_factor,
+                    default_margin_ratio,
+                    MARGIN_PRECISION_U128,
+                    true,
+                )?;
+
+                default_margin_ratio.max(size_adj_margin_ratio)
```

Comment:
is this redundant since you pass capped to true? same for sdk

Reply:
nah thats a different thing in that function was previously just always true